src/cow-react/modules/limitOrders/containers/OrdersWidget/index.tsx

```diff
@@ -37,14 +42,20 @@ export function OrdersWidget() {
   }, [ordersList, currentTabId])
 
   const tabs = useMemo(() => {
-    return TABS.map((tab, index) => {
+    return TABS.map((tab) => {
       return { ...tab, isActive: tab.id === currentTabId, count: getOrdersListByIndex(ordersList, tab.id).length }
     })
   }, [currentTabId, ordersList])
 
+  // Check balances and allowances only for pending orders
+  const balancesAndAllowances =
+    currentTabId === openTab.id ? pendingBalancesAndAllowances : { balances: {}, allowances: {} }
```

Comment:
Actually this filters the balance/allowance check only at display time, but they are still being queried in the background.
What about making it so the query doesn't trigger if not in the right tab?

Reply:
Thanks! Good point. FIxed